benchmark/utils_sr.js

```diff
@@ -0,0 +1,172 @@
+    document.getElementById('image').style.display = 'none';
+    this.inCanvas.width = this.modelInfoDict.inputSize[1];
+    this.inCanvas.height = this.modelInfoDict.inputSize[0];
+    this.outCanvas.width = this.modelInfoDict.outputSize[1];
+    this.outCanvas.height = this.modelInfoDict.outputSize[0];
+    this.inWidth = this.webWidth;
+    this.inHeight = this.inWidth * (this.modelInfoDict.inputSize[0] / this.modelInfoDict.inputSize[1]);
+    if (this.inHeight > this.webHeight) {
+      this.inHeight = this.webHeight;
+      this.inWidth = this.inHeight * (this.modelInfoDict.inputSize[1] / this.modelInfoDict.inputSize[0]);
+    }
+    this.outWidth = this.webWidth;
+    this.outHeight = this.outWidth * (this.modelInfoDict.outputSize[0] / this.modelInfoDict.outputSize[1]);
+    if (this.outHeight > this.webHeight) {
+      this.outHeight = this.webHeight;
+      this.outWidth = this.outHeight * (this.modelInfoDict.outputSize[1] / this.modelInfoDict.outputSize[0]);
+    }
+    canvasElement.setAttribute("width", this.inWidth);
+    canvasElement.setAttribute("height", this.inHeight);
+    this.outputElement.setAttribute("width", this.outWidth);
+    this.outputElement.setAttribute("height", this.outHeight);
+    canvasElement.getContext('2d').drawImage(imageElement, 0, 0, this.inWidth, this.inHeight);
+  }
+
+  async setInputOutput() {
+    this.setInputOutputUI();
+    let inWidth = this.modelInfoDict.inputSize[1];
+    let inHeight = this.modelInfoDict.inputSize[0];
+    const channels = this.modelInfoDict.inputSize[2];
+    const imageChannels = 4; // RGBA
+    const [mean, offset] = [127.5, 1];
+    this.isQuantized = this.modelInfoDict.isQuantized || false;
+    let typedArray;
+    if (this.isQuantized) {
+      typedArray = Uint8Array;
+    } else {
+      typedArray = Float32Array;
+    }
+    if (bkImageSrc === null) {
+      bkImageSrc = imageElement.src;
+    } else {
+      imageElement.src = bkImageSrc;
+    }
+    this.inputTensor = new typedArray(this.modelInfoDict.inputSize.reduce((a, b) => a * b));
+    this.outputTensor = new typedArray(this.modelInfoDict.outputSize.reduce((a, b) => a * b));
+    this.inCtx.drawImage(imageElement, 0, 0, inWidth, inHeight);
+    let pixels = this.inCtx.getImageData(0, 0, inWidth, inHeight).data;
+    for (let y = 0; y < inHeight; ++y) {
+      for (let x = 0; x < inWidth; ++x) {
+        for (let c = 0; c < channels; ++c) {
+          let value = pixels[y * inWidth * imageChannels + x * imageChannels + c];
+          this.inputTensor[y * inWidth * channels + x * channels + c] = value / mean - offset;
+        }
+      }
+    }
+  }
+
+  /**
+   * Setup model
+   * @returns {Promise<void>}
+   */
+  async setupAsync() {
+    await this.setInputOutput();
+    let backend = this.backend.replace('WebNN', 'WebML');
+    const modelName = this.modelInfoDict.modelName;
+    let loadResult = await loadModelAndLabels(this.modelInfoDict.modelFile, null);
```

Comment:
@cuiyanx  I suggest to update function loadModelAndLabels using default parameter label=null, what do you think？

Reply:
@BruceDai Agree!